homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -338,30 +352,31 @@ def _build_observations_by_entity(self):
         Build and return data structure of the form below.
 
         {
-            "sensor.sensor1": [{"id": 0, ...}, {"id": 1, ...}],
-            "sensor.sensor2": [{"id": 2, ...}],
+            "sensor.sensor1": [Observation, Observation],
+            "sensor.sensor2": [Observation],
             ...
         }
 
         Each "observation" must be recognized uniquely, and it should be possible
         for all relevant observations to be looked up via their `entity_id`.
         """
 
-        observations_by_entity: dict[str, list[OrderedDict]] = {}
-        for i, obs in enumerate(self._observations):
-            obs["id"] = i
+        observations_by_entity: dict[str, list[Observation]] = {}
+        for _, observation in enumerate(self._observations):
```

Comment:
Can you drop `enumerate` here?
```suggestion
        for observation in self._observations:
```